src/foam/nanos/auth/Group.js

```diff
@@ -66,19 +66,22 @@ foam.CLASS({
           javaType: 'java.security.Permission'
         }
       ],
-      javaCode: `
-        if ( getPermissions() == null ) return false;
-        for ( int i = 0 ; i < permissions_.length ; i++ )
-          if ( new javax.security.auth.AuthPermission(permissions_[i].getId()).implies(permission) )
+      javaCode:
+        `if ( getPermissions() == null ) return false;
+        for ( int i = 0 ; i < permissions_.length ; i++ ) {
+          if ( new javax.security.auth.AuthPermission(permissions_[i].getId()).implies(permission) ) {
             return true;
-        return false;
-      `,
-      code: `
-        if ( getPermissions() == null ) return false;
-        for ( int i = 0 ; i < permissions_.length ; i++ )
-          if ( permissions_[i].implies(permission) ) return true;
-        return false;
-      `
+          }
+        }
+        return false;`
+      ,
+      code: function(permissionId, permission, group) {
+        if ( this.permissions == null ) return false;
+
+        for ( var i = 0 ; i < this.permissions.length ; i++ ) {
+          return group.permissions[i].implies(group.permissions[i].id, permissionId);
```

Comment:
Why are you passing group.permission[i].id to group.permission[i].implies()?, since it's the same object, it already knows its own .id, so doesn't need to be told.